### PR TITLE
[ID-2146] chore: Add tracking metrics to root Passport class

### PR DESCRIPTION
**Pull request by matthewmuscat**
# Summary
https://immutable.atlassian.net/browse/ID-2146

Adds further observability around root Passport SDK methods.

# Detail and impact of the change

## Security

# Anything else worth calling out?

### CODE REVIEW

File: packages/passport/sdk/src/Passport.ts [hunk 1]
```diff
@@ -134,18 +135,29 @@ export class Passport {
...
+    flow.addEvent('endConnectImxSilent');
+    return provider;
   }
 
   public async connectImx(): Promise<IMXProvider> {
-    return this.passportImxProviderFactory.getProvider();
+    const flow = trackFlow('passport', 'connectImx');
+    flow.addEvent('startConnectImx');
+    const provider = this.passportImxProviderFactory.getProvider();
+    flow.addEvent('endConnectImx');
```
Comment (haydenfowler) [inline packages/passport/sdk/src/Passport.ts:149]: Can we follow the same pattern that we're using in `zkEvmProvider.ts` please? E.g.:
```
const flow = trackFlow('passport', FLOW_NAME);
try {
} catch (error) {
  if (error instanceof Error) {
    trackError('passport', FLOW_NAME, error);
  }
  flow.addEvent('errored');
  throw error;
} finally { 
  flow.addEvent('end');
}
```

File: packages/passport/sdk/src/Passport.ts [hunk 2]
```diff
@@ -134,35 +135,74 @@ export class Passport {
    * `connectImx` instead.
    */
   public async connectImxSilent(): Promise<IMXProvider | null> {
-    return this.passportImxProviderFactory.getProviderSilent();
+    const flow = trackFlow('passport', 'connectImxSilent');
+
+    try {
+      flow.addEvent('startConnectImxSilent');
```
Comment (haydenfowler) [inline packages/passport/sdk/src/Passport.ts:141]: The `start` event seems to be triggered when `trackFlow` is called (see [here](https://github.com/immutable/ts-immutable-sdk/blob/d4aa5e257dbdf681cdd241decbe689f891ca4476/packages/internal/metrics/src/flow.ts#L104)), so we should be able to omit all of these `flow.addEvent('startXXXXX');` calls

**Review by haydenfowler — COMMENTED** [review]
LGTM 👍 happy to approve once the "start" events are removed

File: packages/passport/sdk/src/Passport.ts (hunk 2 above)
Comment (matthewmuscat) [reply packages/passport/sdk/src/Passport.ts:141]: Ahh good catch, thanks!

**Review by haydenfowler — DISMISSED** [review]

File: packages/passport/sdk/src/utils/metrics.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { trackError, trackFlow } from '@imtbl/metrics';
+
+export const withMetrics = <T>(
+  fn: () => T,
+  event: string,
+): T => {
+  const flow = trackFlow('passport', event);
+
+  try {
+    return fn();
```
Comment (haydenfowler) [inline packages/passport/sdk/src/utils/metrics.ts:10]: LGTM 👍 
We may want to extend these implementations in the future to pass the flow to the function being executed, e.g:

```suggestion
  fn: (flow: Flow) => T,
  event: string,
): T => {
  const flow = trackFlow('passport', event);

  try {
    return fn(flow);
```

**Review by haydenfowler — APPROVED** [review]

**Pull request merged**